grid-cli/cmd/deploy_vm.go

```diff
@@ -152,40 +130,48 @@ var deployVMCmd = &cobra.Command{
 			log.Fatal().Err(err).Send()
 		}
 
-		if node == 0 {
-			filter, ssd, rootfss := filters.BuildVMFilter(vm, diskMount, volumeMount, farm)
-			nodes, err := deployer.FilterNodes(
-				cmd.Context(),
-				t,
-				filter,
-				ssd,
-				nil,
-				rootfss,
-			)
-			if err != nil {
+		// if no public ips or yggdrasil then we should go for the light deployment
+		if !ipv4 && !ipv6 && !ygg {
+			vm := workloads.VMLight{
+				Name:           name,
+				EnvVars:        env,
+				CPU:            cpu,
+				MemoryMB:       memory * 1024,
+				GPUs:           convertGPUsToZosGPUs(gpus),
+				RootfsSizeMB:   rootfs * 1024,
+				Flist:          flist,
+				Entrypoint:     entrypoint,
+				MyceliumIPSeed: seed,
+			}
+			err = executeVMLight(cmd.Context(), t, vm, node, farm, disk, volume)
+			if err == nil {
+				return nil
+			}
+
+			if err != nil && !errors.Is(err, errNoNodesFound) {
```

Comment:
the error is guaranteed to not be nil here so you need just to onle check if `!errors.Is(err, errNoNodesFound)`